packages/postgres-database/tests/projects/test_projects_version_control.py

```diff
@@ -0,0 +1,285 @@
+# pylint: disable=redefined-outer-name
+# pylint: disable=unused-argument
+# pylint: disable=unused-variable
+# pylint: disable=too-many-statements
+
+import hashlib
+import json
+from typing import Any, Dict, Optional
+from uuid import UUID, uuid3
+
+import pytest
+from aiopg.sa.connection import SAConnection
+from aiopg.sa.result import RowProxy
+from simcore_postgres_database.models.projects import projects
+from simcore_postgres_database.models.projects_version_control import (
+    projects_vc_branches,
+    projects_vc_commits,
+    projects_vc_heads,
+    projects_vc_repos,
+    projects_vc_snapshots,
+    projects_vc_tags,
+)
+from simcore_postgres_database.utils_aiopg_orm import BaseOrm
+
+
+class ReposOrm(BaseOrm[int]):
+    def __init__(self, connection: SAConnection):
+        super().__init__(
+            projects_vc_repos,
+            connection,
+            readonly={"id", "created", "modified"},
+        )
+
+
+class BranchesOrm(BaseOrm[int]):
+    def __init__(self, connection: SAConnection):
+        super().__init__(
+            projects_vc_branches,
+            connection,
+            readonly={"id", "created", "modified"},
+        )
+
+
+class CommitsOrm(BaseOrm[int]):
+    def __init__(self, connection: SAConnection):
+        super().__init__(
+            projects_vc_commits,
+            connection,
+            readonly={"id", "created", "modified"},
+            # pylint: disable=no-member
+            writeonce=set(c for c in projects_vc_commits.columns.keys()),
+        )
+
+
+class TagsOrm(BaseOrm[int]):
+    def __init__(self, connection: SAConnection):
+        super().__init__(
+            projects_vc_tags,
+            connection,
+            readonly={"id", "created", "modified"},
+        )
+
+
+class ProjectsOrm(BaseOrm[str]):
+    def __init__(self, connection: SAConnection):
+        super().__init__(
+            projects,
+            connection,
+            readonly={"id", "creation_date", "last_change_date"},
+            writeonce={"uuid"},
+        )
+
+
+class SnapshotsOrm(BaseOrm[str]):
+    def __init__(self, connection: SAConnection):
+        super().__init__(
+            projects_vc_snapshots,
+            connection,
+            writeonce={"checksum"},  # TODO:  all? cannot delete snapshots?
+        )
+
+
+class HeadsOrm(BaseOrm[int]):
+    def __init__(self, connection: SAConnection):
+        super().__init__(
+            projects_vc_heads,
+            connection,
+            writeonce={"repo_id"},
+        )
+
+
+# -------------
+
+
+def eval_checksum(workbench: Dict[str, Any]):
+    # FIXME: prototype
+    block_string = json.dumps(workbench, sort_keys=True).encode("utf-8")
+    raw_hash = hashlib.sha256(block_string)
+    return raw_hash.hexdigest()
+
+
+def eval_snapshot_uuid(repo: RowProxy, commit: RowProxy) -> UUID:
+    assert repo.id == commit.repo_id  # nosec
+    return uuid3(UUID(repo.project_uuid), f"{repo.id}.{commit.snapshot_checksum}")
```

Comment:
why not uuid5?

Reply:
why not... :-) The different between uuid3 and 5 is that the first uses MD5 and the second SHA-1 

As I said above, how to compute the checksum, snapshot and eventually the uuid is defined externally. Here in the tests, I have added this an example. 

That said, when I implement this in the next PR, i will use uuid5